### PR TITLE
Clean pkg before vet

### PR DESCRIPTION
**Pull request by exekias**
This ensure we ignore cached compiled packages when doing vet, some
changes break type checking from the previous version.

In go 1.9 vet has a `-source` flag. From vet doc at https://golang.org/cmd/vet/:

  By default vet uses the object files generated by 'go install
  some/pkg' to typecheck the code. If the -source flag is provided, vet
  uses only source code.`

### CODE REVIEW

File: libbeat/scripts/Makefile [hunk 1]
```diff
@@ -103,6 +103,7 @@ crosscompile: $(GOFILES)
 
 .PHONY: check
 check: python-env ## @build Checks project and source code if everything is according to standard
+	@rm -rf ${GOPATH}/pkg # TODO go 1.9 supports `-source` flag in go vet, remove this in favor of the flag
```
Comment (ruflin) [inline libbeat/scripts/Makefile]: It's quiet intrusive that we remove the complete `pkg` folder. I wonder if this could have some side affects.

**Comment by exekias** [comment]
jenkins retest this please

File: libbeat/scripts/Makefile (hunk 1 above)
Comment (exekias) [reply libbeat/scripts/Makefile]: I'll change this to go/pkg/linux_amd64/github.com/elastic/beats/ only, does it work for you?

**Comment by exekias** [comment]
we finally decided to put this in jenkins, where it belongs

**Pull request closed without merging**